src/ucp/tag/offload.c

```diff
@@ -299,15 +299,20 @@ ucp_tag_offload_do_post(ucp_request_t *req)
                                         req->recv.length, req->recv.datatype,
                                         &req->recv.state, req->recv.mem_type,
                                         req, UCT_MD_MEM_FLAG_HIDE_ERRORS);
-        if ((status != UCS_OK) || !req->recv.state.dt.contig.md_map) {
+        if (status != UCS_OK) {
```

Comment:
could we return error in case of UCT_MD_MEM_FLAG_HIDE_ERRORS ?
change to ucs_assertv(status == UCS_OK)
then no need to use goto since block_mem_reg will be used only once